pom.xml

```diff
@@ -77,13 +77,19 @@
     <build>
         <resources>
             <resource>
-                <directory>src/main/assembly/dist</directory>
+                <directory>src/main/resources</directory>
                 <targetPath>${project.build.directory}/version</targetPath>
                 <includes>
                     <include>version</include>
                 </includes>
                 <filtering>true</filtering>
             </resource>
+            <resource>
```

Comment:
why do you need this extra resource to exclude `version` when you included it above?

Reply:
In the topmost **resource** `version` is written into `${project.build.directory}/version`.  In the **resource** under it all other entitities from `src/main/resources` are included and written into another directory.  To avoid `version` to be written a second time it has to be excluded here.